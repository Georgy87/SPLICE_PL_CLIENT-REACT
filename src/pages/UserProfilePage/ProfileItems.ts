export const userInfoItems = {
	Photo: 'Photo',
	Username: 'Username',
	Email: 'Email',
	Name: 'Name',
};

export type UserInfoItems = {
	itemName: keyof typeof userInfoItems;
};

export const ProfileItems: UserInfoItems[] = [
	{ itemName: 'Photo' },
	{ itemName: 'Username' },
	{ itemName: 'Email' },
	{ itemName: 'Name' },
];



export const userInfoTriggers = {
	Update: 'Update',
	DownLoad: 'DownLoad',
	Packs: 'Packs',
};

export type UserInfoTriggers = {
	itemName: keyof typeof userInfoTriggers;
};

export const ProfileTriggerItems: UserInfoTriggers[] = [
	{ itemName: 'Update' },
	{ itemName: 'DownLoad' },
	{ itemName: 'Packs' },
];
