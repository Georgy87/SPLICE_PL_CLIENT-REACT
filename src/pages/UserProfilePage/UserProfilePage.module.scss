.root {
    width: 700px;
}

.userInfoList {
    margin-top: 100px;
}