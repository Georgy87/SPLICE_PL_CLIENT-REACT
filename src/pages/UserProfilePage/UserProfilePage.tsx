import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';

import { UserProfileItem } from '../../components/UserProfileItem/UserProfileItem';
import { ButtonLayout } from '../../layouts/ButtonLayout/ButtonLayout';
import { selectUser } from '../../store/selectors/userSelectors';
import { fetchUpdateEmail, fetchUpdateFullName } from '../../store/slices/user/userSlice';
import { ProfileItems } from './ProfileItems';

import styles from './UserProfilePage.module.scss';

export const UserProfilePage: React.FC = () => {
	const history = useHistory();
	const dispatch = useDispatch();

	const user = useSelector(selectUser);

	const [email, setEmail] = useState<string | undefined>('');
	const [fullname, setFullName] = useState<string | undefined>('');

	const onUpdateInfo = () => {
		if (user?.email != email) {
			console.log(user?.email, email);
			dispatch(fetchUpdateEmail({ email }));
		}

		if (user?.fullname != fullname) {
			dispatch(fetchUpdateFullName({ fullname }));
		}
	};

	useEffect(() => {
		setEmail(user?.email);
	}, [user?.email]);

	useEffect(() => {
		setFullName(user?.fullname);
	}, [user?.fullname]);

	return (
		<div className={styles.root}>
			<ul className={styles.userInfoList}>
				{ProfileItems.map((profileItems: ProfileItems, idx: number) => {
					return (
						<UserProfileItem
							key={idx}
							profileItems={profileItems.itemName}
							setFullName={setFullName}
							setEmail={setEmail}
						/>
					);
				})}
				<div className={styles.downloadPack}>
					<h1>Update</h1>
					<ButtonLayout typeStyle='update' onClicked={onUpdateInfo}>
						Update
					</ButtonLayout>
				</div>

				<div className={styles.downloadPack}>
					<h1>DOWNLOAD PACK</h1>
					<ButtonLayout
						onClicked={() => history.push('/profile/create')}
						typeStyle='update'
					>
						Download
					</ButtonLayout>
				</div>
			</ul>
		</div>
	);
};
