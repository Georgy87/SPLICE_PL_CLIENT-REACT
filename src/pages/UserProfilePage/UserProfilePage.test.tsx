import { render } from '@testing-library/react';

import { renderTest } from '../../tests/helpers/RenderTestApp';
import { UserProfilePage } from './UserProfilePage';

describe('UserProfilePage component', () => {
	test('userProfilePage snapshot', async () => {
		const userProfilePage = render(renderTest(<UserProfilePage />));
		expect(userProfilePage).toMatchSnapshot();
	});
});
