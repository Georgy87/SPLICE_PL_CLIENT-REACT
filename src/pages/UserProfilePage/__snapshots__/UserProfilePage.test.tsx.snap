// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPacksPage component UserPacksPage snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="root"
        data-testid="user-profile-page"
      >
        <ul
          class="userInfoList"
        >
          <li
            class="profileListItem"
          >
            <label>
              Photo
            </label>
            <div
              class="avatar"
            >
              <img
                alt="unnamed.jpg"
                src="unnamed.jpg"
              />
            </div>
          </li>
          <li
            class="profileListItem"
          >
            <label>
              Username
            </label>
            <p />
          </li>
          <li
            class="profileListItem"
          >
            <label>
              Email
            </label>
            <input
              value=""
            />
          </li>
          <li
            class="profileListItem"
          >
            <label>
              Name
            </label>
            <input
              value=""
            />
          </li>
          <div
            class="downloadPack"
          >
            <h1>
              Update
            </h1>
            <div
              class="platformBtn"
            >
              <button
                class="platformBtn update"
              >
                Update
              </button>
            </div>
          </div>
          <div
            class="downloadPack"
          >
            <h1>
              DownLoad
            </h1>
            <a
              data-testid="create-pack-link"
              href="/profile/create"
            >
              <div
                class="platformBtn"
              >
                <button
                  class="platformBtn update"
                >
                  DownLoad
                </button>
              </div>
            </a>
          </div>
          <div
            class="downloadPack"
          >
            <h1>
              Packs
            </h1>
            <a
              data-testid="user-packs-link"
              href="/profile/packs"
            >
              <div
                class="platformBtn"
              >
                <button
                  class="platformBtn update"
                >
                  Packs
                </button>
              </div>
            </a>
          </div>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="root"
      data-testid="user-profile-page"
    >
      <ul
        class="userInfoList"
      >
        <li
          class="profileListItem"
        >
          <label>
            Photo
          </label>
          <div
            class="avatar"
          >
            <img
              alt="unnamed.jpg"
              src="unnamed.jpg"
            />
          </div>
        </li>
        <li
          class="profileListItem"
        >
          <label>
            Username
          </label>
          <p />
        </li>
        <li
          class="profileListItem"
        >
          <label>
            Email
          </label>
          <input
            value=""
          />
        </li>
        <li
          class="profileListItem"
        >
          <label>
            Name
          </label>
          <input
            value=""
          />
        </li>
        <div
          class="downloadPack"
        >
          <h1>
            Update
          </h1>
          <div
            class="platformBtn"
          >
            <button
              class="platformBtn update"
            >
              Update
            </button>
          </div>
        </div>
        <div
          class="downloadPack"
        >
          <h1>
            DownLoad
          </h1>
          <a
            data-testid="create-pack-link"
            href="/profile/create"
          >
            <div
              class="platformBtn"
            >
              <button
                class="platformBtn update"
              >
                DownLoad
              </button>
            </div>
          </a>
        </div>
        <div
          class="downloadPack"
        >
          <h1>
            Packs
          </h1>
          <a
            data-testid="user-packs-link"
            href="/profile/packs"
          >
            <div
              class="platformBtn"
            >
              <button
                class="platformBtn update"
              >
                Packs
              </button>
            </div>
          </a>
        </div>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UserProfilePage component userProfilePage snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="root"
        data-testid="user-profile-page"
      >
        <ul
          class="userInfoList"
        >
          <li
            class="profileListItem"
          >
            <label>
              Photo
            </label>
            <div
              class="avatar"
            >
              <img
                alt="unnamed.jpg"
                src="unnamed.jpg"
              />
            </div>
          </li>
          <li
            class="profileListItem"
          >
            <label>
              Username
            </label>
            <p />
          </li>
          <li
            class="profileListItem"
          >
            <label>
              Email
            </label>
            <input
              value=""
            />
          </li>
          <li
            class="profileListItem"
          >
            <label>
              Name
            </label>
            <input
              value=""
            />
          </li>
          <div
            class="downloadPack"
          >
            <h1>
              Update
            </h1>
            <div
              class="platformBtn"
            >
              <button
                class="platformBtn update"
              >
                Update
              </button>
            </div>
          </div>
          <div
            class="downloadPack"
          >
            <h1>
              DownLoad
            </h1>
            <a
              data-testid="create-pack-link"
              href="/profile/create"
            >
              <div
                class="platformBtn"
              >
                <button
                  class="platformBtn update"
                >
                  DownLoad
                </button>
              </div>
            </a>
          </div>
          <div
            class="downloadPack"
          >
            <h1>
              Packs
            </h1>
            <a
              data-testid="user-packs-link"
              href="/profile/packs"
            >
              <div
                class="platformBtn"
              >
                <button
                  class="platformBtn update"
                >
                  Packs
                </button>
              </div>
            </a>
          </div>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="root"
      data-testid="user-profile-page"
    >
      <ul
        class="userInfoList"
      >
        <li
          class="profileListItem"
        >
          <label>
            Photo
          </label>
          <div
            class="avatar"
          >
            <img
              alt="unnamed.jpg"
              src="unnamed.jpg"
            />
          </div>
        </li>
        <li
          class="profileListItem"
        >
          <label>
            Username
          </label>
          <p />
        </li>
        <li
          class="profileListItem"
        >
          <label>
            Email
          </label>
          <input
            value=""
          />
        </li>
        <li
          class="profileListItem"
        >
          <label>
            Name
          </label>
          <input
            value=""
          />
        </li>
        <div
          class="downloadPack"
        >
          <h1>
            Update
          </h1>
          <div
            class="platformBtn"
          >
            <button
              class="platformBtn update"
            >
              Update
            </button>
          </div>
        </div>
        <div
          class="downloadPack"
        >
          <h1>
            DownLoad
          </h1>
          <a
            data-testid="create-pack-link"
            href="/profile/create"
          >
            <div
              class="platformBtn"
            >
              <button
                class="platformBtn update"
              >
                DownLoad
              </button>
            </div>
          </a>
        </div>
        <div
          class="downloadPack"
        >
          <h1>
            Packs
          </h1>
          <a
            data-testid="user-packs-link"
            href="/profile/packs"
          >
            <div
              class="platformBtn"
            >
              <button
                class="platformBtn update"
              >
                Packs
              </button>
            </div>
          </a>
        </div>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
